### PR TITLE
Update mongodb to 4.4/5.0

### PR DESCRIPTION
**Pull request by kfiz**
###### Motivation for this change
Adds new mongodb-4_4 and mongodb-5_0 targets

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by otavio** [comment]
@kfiz please:

- rebase on top of master
- update 4.4 to 4.4.8
- update 5.0 to 5.0.2

File: pkgs/development/libraries/boost/1.76.nix [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  version = "1.76.0";
+
+  src = fetchurl {
+    urls = [
+      "mirror://sourceforge/boost/boost_1_76_0.tar.bz2"
+      "https://dl.bintray.com/boostorg/release/1.76.0/source/boost_1_76_0.tar.bz2"
+    ];
+    # SHA256 from http://www.boost.org/users/history/version_1_76_0.html
+    sha256 = "0hcc661savk32hx65997p0ss1awj6ala4cmz4w7lbi42x6k7nfgh";
+  };
```
Comment (otavio) [inline pkgs/development/libraries/boost/1.76.nix:13]: Is this new boost release required?

If yes, please do it in another commit prior to the mongo changes; if not, please split it in another PR.

**Review by otavio — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/boost/1.76.nix (hunk 1 above)
Comment (kfiz) [reply pkgs/development/libraries/boost/1.76.nix:13]: Yes this is necessary as it is explicitly stated as requirement in downloaded archive. I've separated this update into a separate commit.

**Review by otavio — APPROVED** [review]

File: pkgs/servers/nosql/mongodb/v5_0.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{ stdenv, callPackage, lib, sasl, boost, Security, CoreFoundation, cctools }:
+
+let
+  buildMongoDB = callPackage ./mongodb.nix {
+    inherit sasl;
+    inherit boost;
+    inherit Security;
+    inherit CoreFoundation;
+    inherit cctools;
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/v5_0.nix:9]:
```suggestion
    inherit sasl boost Security CoreFoundation cctools;
```

File: pkgs/servers/nosql/mongodb/v5_0.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  };
+in
+buildMongoDB {
+  version = "5.0.2";
+  sha256 = "05g3ch0dnn9qf7xd2l1i2fq7m20hhz1qpvd7bri0khk8fm744xw8";
+  patches =
+    [ ./forget-build-dependencies-4-4.patch ]
+    ++ lib.optionals stdenv.isDarwin [ ./asio-no-experimental-string-view-4-4.patch ];
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/v5_0.nix:17]:
```suggestion
  patches = [
    ./forget-build-dependencies-4-4.patch
  ] ++ lib.optionals stdenv.isDarwin [ ./asio-no-experimental-string-view-4-4.patch ];
```

File: pkgs/servers/nosql/mongodb/v4_4.nix [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    inherit CoreFoundation;
+    inherit cctools;
+  };
+in
+buildMongoDB {
+  version = "4.4.8";
+  sha256 = "15saa2xfkr5xz18kbz64k7g3d6a9pxv8bzsij36w67pndridd6vg";
+  patches =
+    [ ./forget-build-dependencies-4-4.patch ]
+    ++ lib.optionals stdenv.isDarwin [ ./asio-no-experimental-string-view-4-4.patch ];
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/v4_4.nix:17]:
```suggestion
  patches = [
    ./forget-build-dependencies-4-4.patch
  ] ++ lib.optionals stdenv.isDarwin [ ./asio-no-experimental-string-view-4-4.patch ];
```

File: pkgs/servers/nosql/mongodb/v4_4.nix [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{ stdenv, callPackage, lib, sasl, boost, Security, CoreFoundation, cctools }:
+
+let
+  buildMongoDB = callPackage ./mongodb.nix {
+    inherit sasl;
+    inherit boost;
+    inherit Security;
+    inherit CoreFoundation;
+    inherit cctools;
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/v4_4.nix:9]:
```suggestion
    inherit sasl boost Security CoreFoundation cctools;
```

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 6]
```diff
@@ -1,27 +1,50 @@
...
+, boost
+, gperftools
+, pcre-cpp
+, snappy
+, zlib
+, libyamlcpp
+, sasl
+, openssl
+, libpcap
+, python27
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/mongodb.nix:14]: Can we get rid of python2?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 6 above)
Comment (kfiz) [reply pkgs/servers/nosql/mongodb/mongodb.nix:14]: Haven't touched that part, but it does seem that it is redundant as it is not used anywhere in the rest of the code. I'll remove it.

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 7]
```diff
@@ -1,27 +1,50 @@
...
+, libyamlcpp
+, sasl
+, openssl
+, libpcap
+, python3
+, curl
+, Security
+, CoreFoundation
+, cctools
+, lzma ? null
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/mongodb.nix:19]:
```suggestion
, lzma
```

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 8]
```diff
@@ -44,7 +68,7 @@ in stdenv.mkDerivation rec {
     inherit sha256;
   };
 
-  nativeBuildInputs = [ variants.scons ];
+  nativeBuildInputs = [ variants.scons ] ++ lib.optionals (stdenv.isLinux && versionAtLeast version "4.4") [ lzma ];
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/mongodb.nix:71]:
```suggestion
  nativeBuildInputs = [ variants.scons ]
    ++ lib.optionals (stdenv.isLinux && versionAtLeast version "4.4") [ lzma ];
```
You are sure this is only needed on Linux?

File: pkgs/top-level/all-packages.nix [hunk 9]
```diff
@@ -4714,6 +4714,8 @@ with pkgs;
 
   endlessh = callPackage ../servers/endlessh { };
 
+  ericw-tools = callPackage ../applications/misc/ericw-tools { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:4717]: There are many changes here which seem to be totally unrelated.

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 8 above)
Comment (kfiz) [reply pkgs/servers/nosql/mongodb/mongodb.nix:71]: Not sure for arm, but for x86_64-darwinthis is not needed.

File: pkgs/top-level/all-packages.nix (hunk 9 above)
Comment (kfiz) [reply pkgs/top-level/all-packages.nix:4717]: Let me rebase again to master. Seems like something went wrong.

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 7 above)
Comment (kfiz) [reply pkgs/servers/nosql/mongodb/mongodb.nix:19]: Fails basic evaluation test if I skip the `null`

Comment (SuperSandro2000) [reply pkgs/servers/nosql/mongodb/mongodb.nix:19]: This means there is something wrong here. lzma is an alias to xz. Either replace lzma with xz or remove it entirely if xz is already part of the inputs/buildInputs.

Comment (kfiz) [reply pkgs/servers/nosql/mongodb/mongodb.nix:19]: Not quite. The lzma/xz dependency is only needed starting in MongoDB 4.4. on x86_64-linux. For earlier versions it is not required thus the option. Is there a more elegant way to do this?

File: pkgs/development/libraries/boost/generic.nix [hunk 10]
```diff
@@ -22,10 +22,10 @@
 , enablePython ? false
 , enableNumpy ? false
 , taggedLayout ? ((enableRelease && enableDebug) || (enableSingleThreaded && enableMultiThreaded) || (enableShared && enableStatic))
-, patches ? [ ]
+, patches ? []
```
Comment (otavio) [inline pkgs/development/libraries/boost/generic.nix]: Unrelated.

File: pkgs/development/libraries/boost/generic.nix [hunk 11]
```diff
@@ -22,10 +22,10 @@
...
 , useMpi ? false
 , mpi
-, extraB2Args ? [ ]
+, extraB2Args ? []
```
Comment (otavio) [inline pkgs/development/libraries/boost/generic.nix:28]: Unrelated.

File: pkgs/development/libraries/boost/generic.nix [hunk 12]
```diff
@@ -120,7 +120,7 @@ stdenv.mkDerivation {
 
   inherit src version;
 
-  patchFlags = [ ];
+  patchFlags = [];
```
Comment (otavio) [inline pkgs/development/libraries/boost/generic.nix:123]: Unrelated.

File: pkgs/development/libraries/boost/generic.nix [hunk 13]
```diff
@@ -129,15 +129,13 @@ stdenv.mkDerivation {
...
-      stripLen = 1;
-      extraPrefix = "libs/context/";
-    })
+    ++ optional (versionAtLeast version "1.61" && versionOlder version "1.71")
+      (fetchpatch {
+        url = "https://github.com/boostorg/context/commit/2354eca9b776a6739112833f64754108cc0d1dc5.patch";
+        sha256 = "067m4bjpmcanqvg28djax9a10avmdwhlpfx6gn73kbqqq70dnz29";
+        stripLen = 1;
+        extraPrefix = "libs/context/";
+      })
```
Comment (otavio) [inline pkgs/development/libraries/boost/generic.nix:138]: Unrelated.

It seems you didn't apply nixfmt.

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 7 above)
Comment (otavio) [reply pkgs/servers/nosql/mongodb/mongodb.nix:19]: I believe you can bring it to scope but add it as input of the versions that is newer than 4.4.

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 14]
```diff
@@ -44,7 +68,9 @@ in stdenv.mkDerivation rec {
     inherit sha256;
   };
 
-  nativeBuildInputs = [ variants.scons ];
+  nativeBuildInputs = [ variants.scons ]
+    ++ lib.optionals (stdenv.isLinux && versionAtLeast version "4.4") [ lzma ];
```
Comment (otavio) [inline pkgs/servers/nosql/mongodb/mongodb.nix:72]: The manual don't state LZMA being platform-specific so please pin it to the version only.

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 15]
```diff
@@ -95,7 +120,8 @@ in stdenv.mkDerivation rec {
     "--use-sasl-client"
     "--disable-warnings-as-errors"
     "VARIANT_DIR=nixos" # Needed so we don't produce argument lists that are too long for gcc / ld
-  ] ++ map (lib: "--use-system-${lib}") system-libraries;
+  ] ++ lib.optionals (versionAtLeast version "4.4") [ "--link-model=static" ]
```
Comment (otavio) [inline pkgs/servers/nosql/mongodb/mongodb.nix:123]: Why using static here?

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 16]
```diff
@@ -119,7 +145,9 @@ in stdenv.mkDerivation rec {
     runHook postInstallCheck
   '';
 
-  prefixKey = "--prefix=";
+  installTargets = if (versionAtLeast version "4.4") then "install-core" else "install";
+
+  prefixKey = if (versionAtLeast version "4.4") then "DESTDIR=" else "--prefix=";
```
Comment (otavio) [inline pkgs/servers/nosql/mongodb/mongodb.nix:150]: Prefix and DESTDIR usually have different meanings. Is this right?

**Review by otavio — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/boost/generic.nix [hunk 17]
```diff
@@ -1,10 +1,18 @@
...
+, python ? null
+, fixDarwinDylibNames
+, libiconv
 , boost-build
 , fetchpatch
 , which
-, toolset ? /**/ if stdenv.cc.isClang  then "clang"
-            else if stdenv.cc.isGNU    then "gcc"
-            else null
+, toolset ? /**/ if stdenv.cc.isClang then "clang"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/boost/generic.nix:13]:
```suggestion
, toolset ? if stdenv.cc.isClang then "clang"
```
If this is being formatted that comment no longer servers a purpose.

**Review by SuperSandro2000 — COMMENTED** [review]
It would be easier if the PR wouldn't contain so much unrelated formatting.

File: pkgs/development/libraries/boost/generic.nix (hunk 17 above)
Comment (kfiz) [reply pkgs/development/libraries/boost/generic.nix:13]: Yeah, unfortunately I had formatting on save set up in my editor so everything was reformatted in addition to the actual changes. I think we can go with @adrian-gierakowski version of the PR as he has cleaned this up and close this one.

Comment (adrian-gierakowski) [reply pkgs/development/libraries/boost/generic.nix:13]: Yes, sorry guys, I should have opened a PR by now. I’ll do later today

**Comment by hjones2199** [comment]
Would this build with boost 1.77 instead of 1.76? #138309 has finally made its way into master.

**Comment by kfiz** [comment]
> Would this build with boost 1.77 instead of 1.76? #138309 has finally made its way into master.

I would assume so. I'll check later.

**Comment by kfiz** [comment]
> > Would this build with boost 1.77 instead of 1.76? #138309 has finally made its way into master.
> 
> I would assume so. I'll check later.

Works with boost 1.77. So we could just point to boost once #138309 makes it to master and boost 1.77 is the default. As far as I'm concerned this is ready to go after a rebase. Happy for further improvements/comments. @adrian-gierakowski @otavio @SuperSandro2000 What do you think?

**Comment by otavio** [comment]
I fully agree. One thing though, please check if there are new releases so when rebasing it is up to date.

**Comment by kfiz** [comment]
> I fully agree. One thing though, please check if there are new releases so when rebasing it is up to date.

I've  checked and 5.0.5 seems to be the latest version. I've made that the default on linux and 5.0.3 the default for the rest, since at least on darwin that is the last version that works with the 10.12 SDK provided by nixpkgs.

**Comment by kfiz** [comment]
> > > Would this build with boost 1.77 instead of 1.76? #138309 has finally made its way into master.
> > 
> > 
> > I would assume so. I'll check later.
> 
> Works with boost 1.77. So we could just point to boost once #138309 makes it to master and boost 1.77 is the default. As far as I'm concerned this is ready to go after a rebase. Happy for further improvements/comments. @adrian-gierakowski @otavio @SuperSandro2000 What do you think?

Just saw that #138309 is already in master. So in theory we could use 1.77 already. But looking at older version of the package it seems to be the case that all of them were specified with the minimum required boost, so I'll just leave everything as it is.

File: pkgs/top-level/all-packages.nix [hunk 18]
```diff
@@ -21435,6 +21436,20 @@ with pkgs;
...
+  mongodb-4_4 = callPackage ../servers/nosql/mongodb/v4_4.nix {
+    sasl = cyrus_sasl;
+    boost = boost170.override { enableShared = false; };
+    inherit (darwin) cctools;
+    inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
+  };
+
+  mongodb-5_0 = callPackage ../servers/nosql/mongodb/v5_0.nix {
+    sasl = cyrus_sasl;
+    boost = boost176.override { enableShared = false; };
```
Comment (hjones2199) [inline pkgs/top-level/all-packages.nix:21448]: Is this still necessary? 1.77 is default in master now and per your previous comment mongo is apparently building with 1.77.

Comment (kfiz) [reply pkgs/top-level/all-packages.nix:21448]: Just mentioned this above. Previous versions seem to have always used the minimum required boost, so I wanted to be consistent with that. Happy to change that, but I think we have to keep the override to force static linking.

Comment (hjones2199) [reply pkgs/top-level/all-packages.nix:21448]: I'm of the opinion you should always point to the most recent boost library the program works with, not sure why you would want to use the oldest possible.

Comment (kfiz) [reply pkgs/top-level/all-packages.nix:21448]: No worries. I'll bump to 1.77

**Review by hjones2199 — APPROVED** [review]
LGTM

**Review by otavio — APPROVED** [review]

File: pkgs/development/libraries/boost/1.76.nix [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+})
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/boost/1.76.nix:14]:
```suggestion
{ callPackage, fetchurl, fetchpatch, ... } @ args:

callPackage ./generic.nix (args // rec {
  version = "1.76.0";

  src = fetchurl {
    urls = [
      "mirror://sourceforge/boost/boost_${builtins.replaceStrings ["."] ["_"] version}.tar.bz2"
      "https://dl.bintray.com/boostorg/release/${version}/source/boost_${builtins.replaceStrings ["."] ["_"] version}.tar.bz2"
    ];
    # SHA256 from http://www.boost.org/users/history/version_1_76_0.html
    sha256 = "953db31e016db7bb207f11432bef7df100516eeb746843fa0486a222e3fd49cb";
  };
})
```

updated from version 1.75.

File: pkgs/servers/nosql/mongodb/mongodb.nix [hunk 20]
```diff
@@ -1,5 +1,5 @@
 { lib, stdenv, fetchurl, sconsPackages, boost, gperftools, pcre-cpp, snappy, zlib, libyamlcpp
-, sasl, openssl, libpcap, curl, Security, CoreFoundation, cctools }:
+, sasl, openssl, libpcap, python3, curl, Security, CoreFoundation, cctools, lzma ? null }:
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/mongodb.nix:2]:
```suggestion
, sasl, openssl, libpcap, python3, curl, Security, CoreFoundation, cctools, xz }:
```

File: pkgs/servers/nosql/mongodb/mongodb.nix (hunk 14 above)
Comment (SuperSandro2000) [reply pkgs/servers/nosql/mongodb/mongodb.nix:48]:
```suggestion
    ++ lib.optionals (versionAtLeast version "4.4") [ xz ];
```

File: pkgs/servers/nosql/mongodb/v5_0.nix [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+  buildMongoDB = callPackage ./mongodb.nix {
+    inherit sasl boost Security CoreFoundation cctools;
+  };
+  variants = if stdenv.isLinux
+             then { version = "5.0.5";
+                    sha256 = "1nny7a3w6pk8z9cwsz8kgk7gvf8lh5xb3r24drlr7wv970h90826";
+                  }
+             else { version = "5.0.3"; # at least darwin has to stay on 5.0.3 until the SDK used by nixpkgs is bumped to 10.13
+                    sha256 = "1p9pq0dfd6lynvnz5p1c8dqp4filzrz86j840xwxwx82dm1zl6p0";
+                  };
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/v5_0.nix:13]:
```suggestion
  variants = if stdenv.isLinux then
    {
      version = "5.0.5";
      sha256 = "1nny7a3w6pk8z9cwsz8kgk7gvf8lh5xb3r24drlr7wv970h90826";
    }
  else
    {
      version = "5.0.3"; # at least darwin has to stay on 5.0.3 until the SDK used by nixpkgs is bumped to 10.13
      sha256 = "1p9pq0dfd6lynvnz5p1c8dqp4filzrz86j840xwxwx82dm1zl6p0";
    };
```

File: pkgs/servers/nosql/mongodb/v5_0.nix [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+in
+buildMongoDB {
+  version = variants.version;
+  sha256 = variants.sha256;
+  patches = [
+    ./forget-build-dependencies-4-4.patch
+  ] ++ lib.optionals stdenv.isDarwin [ ./asio-no-experimental-string-view-4-4.patch ];
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/mongodb/v5_0.nix:20]:
```suggestion
    ./asio-no-experimental-string-view-4-4.patch
  ];
```
Or does this break linux?

File: pkgs/top-level/all-packages.nix [hunk 23]
```diff
@@ -21435,6 +21436,20 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
   };
 
+  mongodb-4_4 = callPackage ../servers/nosql/mongodb/v4_4.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21439]:
```suggestion
  mongodb-4_4 = callPackage ../servers/nosql/mongodb/4.4.nix {
```

File: pkgs/top-level/all-packages.nix [hunk 24]
```diff
@@ -21435,6 +21436,20 @@ with pkgs;
...
+    sasl = cyrus_sasl;
+    boost = boost177.override { enableShared = false; };
+    inherit (darwin) cctools;
+    inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
+  };
+
+  mongodb-5_0 = callPackage ../servers/nosql/mongodb/v5_0.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21446]:
```suggestion
  mongodb-5_0 = callPackage ../servers/nosql/mongodb/5.0.nix {
```

File: pkgs/top-level/all-packages.nix (hunk 18 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:21448]:
```suggestion
    boost = boost17x.override { enableShared = false; };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The commits messages do not follow the contributing guide. Please fix that.

File: pkgs/development/libraries/boost/1.76.nix [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+
+  src = fetchurl {
+    urls = [
+      "mirror://sourceforge/boost/boost_${builtins.replaceStrings ["."] ["_"] version}.tar.bz2"
+      "https://dl.bintray.com/boostorg/release/${version}/source/boost_${builtins.replaceStrings ["."] ["_"] version}.tar.bz2"
+    ];
+    # SHA256 from http://www.boost.org/users/history/version_1_76_0.html
+    sha256 = "953db31e016db7bb207f11432bef7df100516eeb746843fa0486a222e3fd49cb";
+  };
+})
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/boost/1.76.nix:14]:
```suggestion
```

File: pkgs/development/libraries/boost/default.nix [hunk 26]
```diff
@@ -44,5 +44,6 @@ in {
   boost173 = makeBoost ./1.73.nix;
   boost174 = makeBoost ./1.74.nix;
   boost175 = makeBoost ./1.75.nix;
+  boost176 = makeBoost ./1.76.nix;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/boost/default.nix:47]:
```suggestion
```

File: pkgs/top-level/all-packages.nix [hunk 27]
```diff
@@ -15921,6 +15921,7 @@ with pkgs;
     boost173
     boost174
     boost175
+    boost176
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15924]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Now we only need to fix the commit messages:
Please do one commit for mongo 4.4 and one for 5.0.

Do we now even need the changes in boost? If not please drop them.

File: pkgs/top-level/all-packages.nix (hunk 27 above)
Comment (kfiz) [reply pkgs/top-level/all-packages.nix:15924]: No, we don't need this. I don't have a preference here. Happy to stay with a gap between 1.75 and 1.77.

**Comment by kfiz** [comment]
> Now we only need to fix the commit messages: Please do one commit for mongo 4.4 and one for 5.0.
> 
> Do we now even need the changes in boost? If not please drop them.

see above. I'll drop it as we don't need it.

File: pkgs/top-level/all-packages.nix (hunk 27 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix]: Ideally most of those versions wouldn't be needed and we could drop a lot of them.

**Comment by kfiz** [comment]
Just tested this again and unfortunately this doesn't build on Darwin due to a setting made in boost that affects the boost 1.77 build on Darwin. I've filed an issue here #162699. Should be an easy fix.

**Comment by bachp** [comment]
While testing this PR I noticed that mongodb no longer builds after a rebase on master. 

The reason seems to be https://jira.mongodb.org/browse/SERVER-59459

Fixes have already landed in all active mongodb branches, so updating to the latest patch versions fixes the issue:

- 4.2.19
- 4.4.13

I think it makes sense to add these updates in this PR to make it testable again.

**Comment by otavio** [comment]
It'd be nice to have this merged soon. MongoDB is lagging behind the releases.

**Comment by bachp** [comment]
@kfiz I did the following:

    rebase on top of master
    update 4.2 to 4.2.19
    update 4.4 to 4.4.13
    update 5.0 to 5.0.7

I hope this is OK for you

**Comment by kfiz** [comment]
> @kfiz I did the following:
> 
> ```
> rebase on top of master
> update 4.2 to 4.2.19
> update 4.4 to 4.4.13
> update 5.0 to 5.0.7
> ```
> 
> I hope this is OK for you

Thanks. That's perfectly fine for me.

**Comment by kfiz** [comment]
> Just tested this again and unfortunately this doesn't build on Darwin due to a setting made in boost that affects the boost 1.77 build on Darwin. I've filed an issue here #162699. Should be an easy fix.

This issue was closed by #168165, which was merged into master.
